lib/aws/ec2.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Created by sonal.ojha on 7/18/2016.
+ */
+var inherits = require('util').inherits;
+var FlowController = require('../flow_controller');
+var AWS = require('aws-sdk');
+
+function AWSEC2() {
+
+    FlowController.call(this);
+
+    var me = this;
+
+    me.findService = function(input) {
+        var params = {region:input.region};
+        if (input.creds)  params.credentials = input.creds;
+        else if (input.profile) {
+            var credentials = new AWS.SharedIniFileCredentials({profile: input.profile});
+            AWS.config.credentials = credentials;
+        }
+        var ec2 = new AWS.EC2(params);
+        return ec2;
+    }
+
+    me.securityGroupHasRules = function(input, callback) {
+        params = {
+            DryRun: false,
+            Filters: [
+                {
+                    Name: 'vpc-id',
+                    Values: [
+                        input.vpcId /* required */
+                    ]
+                },
+                {
+                    Name: 'group-name',
+                    Values: [
+                        input.groupName /* required */
+                    ]
+                }
+            ]
+        };
+        var self = arguments.callee;
+
+        if (callback) {
+            var ec2 = me.findService(input);
+            ec2.describeSecurityGroups(params, callback);
+            return;
+        }
+
+        self.callbackFind = function(data) {
+            var hasRules = false;
+            if (data.SecurityGroups.length !== 0) {
+                for( var item in data.SecurityGroups) {
+                    var permissions = data.SecurityGroups[item].IpPermissions;
+                    console.log("Inbound Rules");
+                    for ( var it in permissions) {
```

Comment:
Yes, we wanted to add whats rules were added in logs so looping through. Actually this changes in awsconfigrules/all_region_vpcs/MA-67 branch.